src/utils/c_utils/src/string_c_utils.c

```diff
@@ -443,3 +443,35 @@ _Bool MallocAndSubstr(char** target, char* source, size_t len)
     *target = t;
     return true;
 }
+
+/**
+ * @brief Produces a new string where each character is replaced with output of @p mapFn call.
```

Comment:
it seems like you're using this to send to tolower(), but there's already strlwr (I think that's it). why reinvent the wheel?

Reply:
I could be wrong, but strlwr (or _strlwr_s) is non-standard, only available in msft c compiler. It might be good to add to ADUCPAL to make it available to non-win platforms.

I don't see strlwr in [FreeBSD string.h](https://cgit.freebsd.org/src/tree/include/string.h?id=c58ff411bc5b19d8fa2ce65fd9d8fabd1b13392f) or in [linux string.h](https://github.com/torvalds/linux/blob/master/include/linux/string.h), but I do see them in msvc crt: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/strlwr-wcslwr?view=msvc-170